bug: undici fails to load for node `>=14 <14.18.0`

## Bug Description

apps with nodejs <14.18.0 have issues loading `undici` because of `node:` prefixed imports.

## Reproducible By

Just start a new project with a simple script

`package.json`
```json
{
  "name": "unidci-node-support",
  "version": "1.0.0",
  "description": "",
  "type": "commonjs",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "undici": "^5.28.1"
  }
}
```

and `index.js`
```javascript
const { request } = require('undici');

request('https://httpstat.us/200')
.then((response) => {
  const {
    statusCode,
    headers,
    trailers,
  } = response;

  console.log('response received', statusCode)
  console.log('headers', headers)
  console.log('trailers', trailers)
});
```

Then run it
```bash
nvm use v1.4.0.0
node index.js
```

## Expected Behavior

The script runs and logs the statusCode, headers & trailers

## Logs & Screenshots

The error
```
internal/modules/cjs/loader.js:1017
  throw err;
  ^

Error: Cannot find module 'node:assert'
Require stack:
- /.../issues/unidci-node-support/node_modules/undici/lib/handler/RetryHandler.js
- /.../issues/unidci-node-support/node_modules/undici/index.js
- /.../issues/unidci-node-support/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1014:15)
    at Function.Module._load (internal/modules/cjs/loader.js:884:27)
    at Module.require (internal/modules/cjs/loader.js:1074:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/Users/david/Documents/repos/issues/unidci-node-support/node_modules/undici/lib/handler/RetryHandler.js:1:16)
    at Module._compile (internal/modules/cjs/loader.js:1185:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1205:10)
    at Module.load (internal/modules/cjs/loader.js:1034:32)
    at Function.Module._load (internal/modules/cjs/loader.js:923:14)
    at Module.require (internal/modules/cjs/loader.js:1074:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/.../issues/unidci-node-support/node_modules/undici/lib/handler/RetryHandler.js',
    '/.../issues/unidci-node-support/node_modules/undici/index.js',
    '/.../issues/unidci-node-support/index.js'
  ]
}
```

## Environment

os: MacOS Sonoma 14.1.1
nodejs: 14.0.0

### Additional context

Issue was found in one of our CI runs https://github.com/elastic/apm-agent-nodejs/pull/3754#issuecomment-1827462112

--- a/lib/handler/RetryHandler.js
+++ b/lib/handler/RetryHandler.js
@@ -1,4 +1,4 @@
-const assert = require('node:assert')
+const assert = require('assert')
 
 const { kRetryHandlerDefaultRetry } = require('../core/symbols')
 const { RequestRetryError } = require('../core/errors')

